routers/repo/pull.go

```diff
@@ -991,3 +993,42 @@ func CleanUpPullRequest(ctx *context.Context) {
 
 	ctx.Flash.Success(ctx.Tr("repo.branch.deletion_success", fullBranchName))
 }
+
+// DownloadPullDiff render a pull's raw diff
+func DownloadPullDiff(ctx *context.Context) {
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Handle(404, "GetIssueByIndex", err)
+		} else {
+			ctx.Handle(500, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	// Redirect elsewhere if it's not a pull request
+	if !issue.IsPull {
+		ctx.Handle(404, "DownloadPullDiff",
+			fmt.Errorf("Issue is not a pull request"))
+		return
+	}
+
+	pr := issue.PullRequest
+
+	if err = pr.GetBaseRepo(); err != nil {
+		ctx.Handle(500, "GetBaseRepo", err)
+		return
+	}
+	if pr.BaseRepo == nil {
+		ctx.Handle(500, "BaseRepo",
```

Comment:
Just move this condition to previous if and you will not need this section. `err != nil || pr.BaseRepo == nil`

Reply:
upon reading the correct GetBaseRepo method I realized there's no way pr.BaseRepo is nil after calling GetBaseRepo and getting no error, so I removed the whole check for null in lattest commit